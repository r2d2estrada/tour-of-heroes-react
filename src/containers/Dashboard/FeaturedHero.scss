.featured-hero {
  cursor: pointer;
  &:hover {
    background-color: var(--primary);
    color: var(--white);
  }
}
