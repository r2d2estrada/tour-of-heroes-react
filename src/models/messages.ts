export interface Message {
  id: number;
  message: string;
}
