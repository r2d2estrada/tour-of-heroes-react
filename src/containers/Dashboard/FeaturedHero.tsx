import React from "react";
import { Hero } from "../../models/hero";
import "./FeaturedHero.scss";

interface FeaturedHeroProps {
  hero: Hero;
}

const FeaturedHero: React.FC<FeaturedHeroProps> = ({ hero }) => {
  return (
    <div className="card featured-hero">
      <div className="card-body">
        <h2 className="card-title text-center">{hero.name}</h2>
      </div>
    </div>
  );
};

export default FeaturedHero;
