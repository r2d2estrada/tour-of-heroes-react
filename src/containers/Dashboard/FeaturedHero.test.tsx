import React from "react";
import { render } from "@testing-library/react";
import FeaturedHero from "./FeaturedHero";
import { Hero } from "../../models/hero";

describe("FeaturedHero", () => {
  const mockHero: Hero = {
    id: 1,
    name: "Test",
  };

  let cpnt: any;

  beforeEach(() => {
    cpnt = <FeaturedHero hero={mockHero} />;
  });

  it("should match snapshot", () => {
    const { container } = render(cpnt);
    expect(container).toMatchSnapshot();
  });
});
