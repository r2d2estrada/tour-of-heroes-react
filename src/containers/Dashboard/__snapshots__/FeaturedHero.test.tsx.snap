// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedHero should match snapshot 1`] = `
<div>
  <div
    class="card featured-hero"
  >
    <div
      class="card-body"
    >
      <h2
        class="card-title text-center"
      >
        Test
      </h2>
    </div>
  </div>
</div>
`;
